lib/api.js

```diff
@@ -498,3 +498,38 @@ exports.update_resources_access_mode_by_ids = function update_resources_access_m
 ) {
   return updateResourcesAccessMode(access_mode, "public_ids[]", ids, callback, options);
 };
+
+exports.create_metadata_field = function create_metadata_field(metadata, callback, options = {}) {
+  options.content_type = "json";
+  return call_api("post", "metadata_fields", metadata, callback, options);
+};
+
+exports.list_metadata_fields = function list_metadata_fields(callback, options = {}) {
+  return call_api("get", "metadata_fields", {}, callback, options);
+};
+
+exports.delete_metadata_field = function delete_metadata_field(external_id, callback, options = {}) {
+  return call_api("delete", "metadata_fields/" + external_id, {}, callback, options);
+};
+
+exports.get_metadata_field = function get_metadata_field(external_id, callback, options = {}) {
+  return call_api("get", "metadata_fields/" + external_id, {}, callback, options);
+};
+
+exports.update_metadata_field = function update_metadata_field(external_id, callback, options = {}) {
+  const params = only(options, "external_id", "type", "label", "mandatory", "default_value", "validation", "datasource");
+  options.content_type = "json";
+  return call_api("put", "metadata_fields/" + external_id, params, callback, options);
+};
+
+exports.update_metadata_field_datasource = function update_metadata_field_datasource(external_id, callback, options = {}) {
+  const params = only(options, "values");
+  options.content_type = "json";
+  return call_api("put", "metadata_fields/" + external_id + "/datasource", params, callback, options);
+};
+
+exports.delete_entries_field_datasource = function delete_entries_field_datasource(external_id, callback, options = {}) {
```

Comment:
Method should be called delete_datasource_entries. Should pass in external_ids as payload rather than taken from options